pkg/transform/kubernetes/kubernetes.go

```diff
@@ -106,6 +106,12 @@ func (t *Transformer) CreateDeployments(o *object.Service) ([]runtime.Object, er
 		},
 	}
 
+	// Set replica value only if it is given
+	if o.Replicas > 0 {
+		replicas := o.Replicas
+		d.Spec.Replicas = &replicas
```

Comment:
@surajssd `d.Spec.Replicas = &o.Replicas` - because you don't take the address of `o` but `o.Replicas`; by being a pointer, it's separate object (memory allocation) so no wasting here